### ISSUE TITLE
Parameter for limiting the amount of itemized deductions to a percent of AGI. 

### ISSUE THREAD

**Issue by MattHJensen** [open-source-economics/Tax-Calculator#1526]
Currently Tax-Calculator users can limit the **tax benefit** of itemized deductions to a percent of AGI. (_ID_BenefitSurtax_*)

CBO also offers a budget option for limiting the amount itemized deduction to a percent of AGI. 

This issue is to add that capability to Tax-Calculator. 

See https://github.com/open-source-economics/Tax-Calculator/issues/1525 for a relevant discussion.

**Comment by MattHJensen** [comment]
I began work on a PR to close this issue, but I ran into a conceptual ambiguity. In particular, I am unsure of how to handle a limit to the amount of a *subset* of itemized deductions. 

When a user wants to limit the _benefit_ from itemized deduction X and itemized deduction Y to a percent of AGI, we calculate taxes with and with out X and Y to calculate the benefit from those two deductions and then tax away the difference between the AGI cap and the benefit. That's easy enough. 

Now suppose a user wants to limit the deductible _amount_ of X and Y. This time we need to limit X and Y _before_ we calculate taxes. In particular, we need to decide how much of the cap will be consumed by X and how much of it by Y. 

If the cap comes out to $100, pre-cap X is $200, and pre-cap Y is $300, what should post-cap X and Y be? 

At first I was inclined to say good riddance and only allow %AGI caps on single deductions or all deductions, but remember that this issue came about because a user wanted to jointly limit a subset of deductions -- (albeit the state and local sales taxes and real estate taxes which are treated the same under current law). 

Now I think we just scale the itemized deductions down proportionally to hit the cap:

post-cap X = cap * X / (X+Y) = $40

Before I implement this, however, I'll wait a few days to see if someone suggests a better route.
